lib/control/include/clingo/control/config.hh

```diff
@@ -0,0 +1,365 @@
+                                  [[maybe_unused]] std::string_view value) {
+            error_("cannot set value");
+        }
+        virtual auto do_array_size() -> std::optional<int> { return std::nullopt; }
+    };
+
+    //! Constructor for ClingoConfig.
+    //!
+    //! @param config a reference to the Clasp CLI configuration object
+    ClingoConfig(Clasp::Cli::ClaspCliConfig &config) : config_{&config} {}
+
+    //! The root key of the configuration.
+    static auto key_root() -> KeyType { return Clasp::Cli::ClaspCliConfig::key_root; }
+
+    //! Retrieves information about the specified key.
+    //!
+    //! @note All out parameters are optional (i.e., can be null).
+    //!
+    //! @param key a handle to a key
+    //! @param n_children the number of subkeys for this key
+    //! @param array_info the length of array keys or -1 for non-array keys
+    //! @param value_info whether the key has a value, can be read, and can be set
+    //! @param help a description of the key
+    void key_info(KeyType key, int *n_children, int *array_info, ValueFlags *value_info) const;
+
+    //! Get the description of a configuration entry.
+    //!
+    //! @param key key of the configuration entry
+    //! @return a string view of the description
+    auto description(KeyType key) const -> std::string_view;
+
+    //! Get the key for an element at a specific index in an array entry.
+    //!
+    //! Returns the key corresponding to the element at the given index in the
+    //! array configuration entry specified by `key`. Throws if the entry is
+    //! not an array or the index is invalid.
+    //!
+    //! @param key key of the array entry
+    //! @param index index of the array element
+    //! @return key for the array element at the given index
+    [[nodiscard]] auto array_at(KeyType key, KeyType index) const -> KeyType;
+
+    //! Get the key for a named subkey in a map entry.
+    //!
+    //! Returns the key corresponding to the subkey with the given path in the
+    //! map configuration entry specified by `key`. Throws if the subkey does
+    //! not exist.
+    //!
+    //! @param key key of the map entry
+    //! @param path path of the subkey
+    //! @return key for the named subkey
+    [[nodiscard]] auto map_at(KeyType key, std::string_view path) const -> std::optional<KeyType>;
+
+    //! Get the name of the nth subkey in a map entry.
+    //!
+    //! Returns the name of the subkey at the given index in the map
+    //! configuration entry specified by `key`. Throws if the index is out of
+    //! bounds.
+    //!
+    //! @param key key of the map entry
+    //! @param index index of the subkey
+    //! @return name of the nth subkey
+    [[nodiscard]] auto map_nth(KeyType key, KeyType index) const -> std::string_view;
+
+    //! Get the value of a configuration entry.
+    //!
+    //! Returns the value associated with the given configuration key.
+    //! Throws if the key does not have a value or cannot be accessed.
```

Comment:
It seems that the implementation actually distinguishes between "does not have a value" and "value cannot be accessed".